crates/frontend/src/workspace/activity_panel/mod.rs

```diff
@@ -162,18 +156,74 @@ impl ActivityPanel {
                     },
                 }
             }))))
-            // this takes up the full height but should only display when there are no activities
-            // and hence no tab bar
             .child_signal(activity_count.signal().map(clone!(height => move |count| {
                 (count == 0).then(|| Self::render_background(height.signal()))
             })))
-            // tabs take up one full line
             .child(column!("is-full", {
                 .class("has-background-white-ter")
                 .child(columns!("is-gapless", "is-mobile", {
                     .children_signal_vec(this.activities.signal_vec_cloned().map(clone!(this => move |activity| {
                         column!("is-narrow", {
                             .child(Activity::render_tab(&activity, &this))
+                            .event_with_options(&EventOptions::preventable(), clone!(context_menu_state => move |event: events::ContextMenu| {
+                                event.prevent_default();  
+                                context_menu_state.show_menu.set(true); 
+                                context_menu_state.menu_position.set((event.x(), event.y())); 
+                            }))
+                            .child_signal(context_menu_state.show_menu.signal_ref(clone!(context_menu_state => move |&show| {
```

Comment:
Use [bool::then](https://doc.rust-lang.org/std/primitive.bool.html#method.then) and [bool::then_some](https://doc.rust-lang.org/std/primitive.bool.html#method.then_some) here.